recipes/libusb/all/conanfile.py

```diff
@@ -61,31 +61,17 @@ def validate(self):
         if self.options.get_safe("enable_udev"):
             if self.settings.os == "Android":
                 raise ConanInvalidConfiguration("udev can't be enabled for Android yet, since libudev recipe is missing in CCI.")
-            if self.settings.os == "Linux" and self._settings_build.os != "Linux":
-                raise ConanInvalidConfiguration("udev can't be enabled yet if cross-compiling to Linux from a non-Linux machine")
+            if tools.cross_building(self.settings):
+                raise ConanInvalidConfiguration("udev can't be enabled yet if cross-compiling")
 
     def build_requirements(self):
         if self._settings_build.os == "Windows" and not self._is_msvc and not tools.get_env("CONAN_BASH_PATH"):
             self.build_requires("msys2/cci.latest")
 
-    def system_requirements(self):
+    def requirements(self):
         if self._settings_build.os == "Linux" and self.settings.os == "Linux":
             if self.options.enable_udev:
```

Comment:
```suggestion
        if self.settings.os == "Linux":
            if self.options.enable_udev:
```

I think that build os check should be delegated to libudev recipe.

Reply:
yes, I think it should remove an option regardless of `settings_build`. to get the same packageID with native and cross-compiling. otherwise, it would be something hard to debug in CI systems (I already had to debug similar case).